### PR TITLE
Update dig version and leverage deference of validation

### PR DESCRIPTION
**Pull request by dnathe4th**
dig@1.5 introduces two significant performance improvements for the container underlying FX. The underlying cycle detection algorithm improvements are most significant for deep, highly-connected dependency graphs. Developers using FX for DI with a significant number of shared components (i.e. a configuration library depended on by most other packages) will see the biggest improvement in startup time.

Additionally, the new `dig.DeferAcyclicVerification()` container constructor option was designed specifically to alleviate redundant safety checks as dependencies as `Provide`d in a tight loop a la FX. 

See https://github.com/uber-go/dig/pull/208 and https://github.com/uber-go/dig/pull/209 for concrete details on the new `dig` features.

### CODE REVIEW

**Comment by glibsm** [comment]
The failed test appears to catch a real issue

**Comment by dnathe4th** [comment]
Hm. I have to disagree that the test catches a real issue. Is there any reason to expect an fx App to be created without an invoke? If so, I need a new strategy. If not, I can update the test so it is not so contrived as to be useless.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Make `fxtest.New` ensure that the app was created successfully. Previously,
   it would return the app (similar to `fx.New`, which expects the user to verify
   the error).
+- Update dig container to defer acyclic validation until after Invoke.
```
Comment (akshayjshah) [inline CHANGELOG.md]: I'd note that this improves performance by reducing repeated work.

File: glide.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ import:
 - package: go.uber.org/multierr
   version: ^1
 - package: go.uber.org/dig
-  version: ^1.4
+  version: ^1.5
```
Comment (akshayjshah) [inline glide.yaml]: We've been bad about this in the past; please leave a comment noting that we need at least v1.5 for the `DeferAcyclicVerification` option.

**Comment by glibsm** [comment]
Dom and I talked offline at length and we agreed that absence of `fx.Invoke` is effectively a no-op and it's ok to allow the cycles at that point (since nothing is used anyways).

We considered other approaches, but none were as simple as "the first invoke will validate".

**Comment by akshayjshah** [comment]
Thanks for the fixes and responding to feedback, Dom! Sorry that it's taken a while - I'll try to merge and cut a release of this tomorrow.

**Pull request merged**